turbopack/crates/turbopack-ecmascript/tests/tree-shaker/analyzer/dce/output.md

```diff
@@ -0,0 +1,224 @@
+# Items
+
+Count: 4
+
+## Item 1: Stmt 0, `ImportOfModule`
+
+```js
+import { baz } from './module';
+
+```
+
+- Hoisted
+- Side effects
+
+## Item 2: Stmt 0, `ImportBinding(0)`
+
+```js
+import { baz } from './module';
+
+```
+
+- Hoisted
+- Declares: `baz`
+
+## Item 3: Stmt 1, `Normal`
+
+```js
+if (1 + 1 == 3) {
+    baz();
+}
+
+```
+
+- Side effects
+- Reads: `baz`
+
+# Phase 1
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+```
+# Phase 2
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+    Item3 --> Item2;
+    Item3 --> Item1;
+```
+# Phase 3
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+    Item3 --> Item2;
+    Item3 --> Item1;
+```
+# Phase 4
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+    Item3 --> Item2;
+    Item3 --> Item1;
+    Item4 --> Item1;
```

Comment:
```suggestion
```

Separate PR:

This edge is not needed. ModuleEvaluation doesn't need to depend on all statements with side effects, it would be enough to depend on the last statement with side effects, since that is referencing the previous one.

Reply:
Filed https://linear.app/vercel/issue/PACK-3280/optimize-tree-shaking